conformance/client/src/main/kotlin/com/connectrpc/conformance/client/adapt/ClientStreamClient.kt

```diff
@@ -46,6 +46,7 @@ abstract class ClientStreamClient<Req : MessageLite, Resp : MessageLite>(
     interface ClientStream<Req : MessageLite, Resp : MessageLite> {
         suspend fun send(req: Req)
         suspend fun closeAndReceive(): ResponseMessage<Resp>
+        suspend fun cancel()
```

Comment:
I had to add this in order to expose the ability to cancel a client-stream RPC. I could have also called it `close` for consistency with other methods. What do you think? Should I change it?

TBH, I kind of _don't like_ that the methods that cancel the RPC are named like "receive close" / "close response", since that's not very clear as to what they're really doing. Maybe when overhauling these interfaces (more-or-less making them look like the interfaces in the conformance client's `adapt` package), we can revisit some of the names and name this operation `cancel` instead of `close` 🤷.

Reply:
I think this is fine for now - maybe there's a better API we could put together that would hook into coroutine cancelation instead of having to expose separate methods for this.

Reply:
I think this is already hooked into coroutine cancellation:
* [unary](https://github.com/connectrpc/connect-kotlin/blob/main/library/src/main/kotlin/com/connectrpc/impl/ProtocolClient.kt#L140-L147)
* [streaming](https://github.com/connectrpc/connect-kotlin/blob/main/library/src/main/kotlin/com/connectrpc/impl/ProtocolClient.kt#L258-L260)

But, I'm not sure I 100% understand exactly how that actually works. I don't think that alone does what we want it to do: the continuation in question for streaming simply returns the bidirectional stream object. So I think the only thing that this would cancel is the actual call to the streaming RPC method. Once it returns the stream object, that continuation is complete and the cancellation no longer does anything.

So I guess we'd have to do something similar in the app code, to tie its coroutine's cancellation to cancelling the stream (which I think means we still need a `cancel` method of some sort). I think it would be challenging to somehow have the framework automatically manage coroutine lifetimes and tie them to the stream since ownership is unclear -- i.e. it's possible that one coroutine could pass the stream to another that runs on a different thread/worker, in which case it's unclear what coroutine cancellation semantics _should_ be.